primitives/io/src/lib.rs

```diff
@@ -842,6 +865,28 @@ pub trait Logging {
 	}
 }
 
+/// Interface that provides functions for profiling the runtime.
+#[runtime_interface]
+pub trait Profiling {
+	/// For enter_span to work correctly with the Tracing profiler,
+	/// the span `target` and `name` must also be registered in `sp_profiler::TracingProfiler`
+	fn enter_span(target: &str, name: &str) -> u64 {
+		PROFILER.lock().create_span(target, name)
```

Comment:
Why don't we forward this to the normal tracer?

Reply:
For 2 reasons it's not that simple, first being that tracing has a requirement that span name and target be `const`. Second is that I wanted to support profiling when the client tracer is not available. Happy to make a compromise though and split things up if needed